google/cloud/security/inventory/README.md

```diff
@@ -44,7 +44,95 @@ $ scripts/dev_inventory.sh
 ```
 
 ## How to Test
-Look at the test instruction in this [README](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/google/cloud/security/README.md#tests).
+Look at the test instruction in this [README].
+
+## How to Add New Resource to Inventory
+
+It's straightforward to add a new resource to inventory for collection.
+This workflow is designed to be generic across resource types.
+
+The main steps are:
+
+1. Define the database table schema to store the resource data.
+
+2. Create a new pipeline worker to process the new table and the table fieldnames.
+
+3. Write a transformation to flatten the data returned by GCP APIs into a format
+that's ingestiable.
+
+Look at this PR for an [end-to-end example loading of a resource].
+
+Step-by-Step:
+
+1. Design a new table schema for the data you need to store in Inventory.
+Each field of data should correspond to a column in the table schema.
+Additionally, we aim to store the raw API data as json to assist
+troubleshooting.  Create a PR for this to be reviewed first before rest of the
+code is written.  See [create_tables.py] for examples.
+
+2. Once the database PR is merged, create a pipeline worker for your resource
+in [inventory/pipelines].  This pipeline worker will then be installed in
+[inventory_loader.py].
+
+3. Get the data from GCP.  [See if the API to get your data already exists.]
+If not, it's your chance to contribute a new one.
+
+4. Data from GCP API is not normalized and can not be fitted into the
+database table.  You will have to write a function to transform the data.
+Please write a [test for this transform function].
+
+    ```json
+    Example API Data
+    {
+        "projects": [
+          {
+            "name": "project1",
+            "parent": {
+              "type": "organization",
+              "id": "888888888888"
+            },
+            "projectId": "project1",
+            "projectNumber": "25621943694",
+            "lifecycleState": "ACTIVE",
+            "createTime": "2016-10-22T16:57:36.096Z"
+          },
+          {
+            "name": "project2",
+            "parent": {
+              "type": "organization",
+              "id": "888888888888"
+            },
+            "projectId": "project2",
+            "projectNumber": "94226340476",
+            "lifecycleState": "ACTIVE",
+            "createTime": "2016-11-13T05:32:10.930Z"
+          },
+        ]
+    }
+
+    Example Transformed (Flattened) Data
+    25621943694 project1  project1  ACTIVE  organization  888888888888  2016-10-22 16:57:36
```

Comment:
Looks nice! Could you add the column names too? (So we can see what the data corresponds to)